tempesta_fw/t/functional/leaks/test_leak.py

```diff
@@ -0,0 +1,110 @@
+"""
+Testing for memory leaks
+"""
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2017 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+import os.path
+import subprocess
+
+import psutil
+from time import sleep
+
+import unittest
+from testers import stress
+from helpers import tf_cfg, control, tempesta
+
+
+class LeakTest(stress.StressTest):
+        """ Leaks testing """
+        config = 'cache 0;\n'
+        backend_connections = 10
+        memory_leak_thresold = 16*1024 # in kib
+
+        def read_kmemleaks(self):
+                """Get amount of kmemleak records"""
+                kmemleakfile = "/sys/kernel/debug/kmemleak"
+                if not os.path.isfile(kmemleakfile):
+                        return -1
+                echocmd = "cat %s | grep \"unreferenced object\" | wc -l" % kmemleakfile
+                proc = subprocess.Popen(["sh", "-c", echocmd], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                result = proc.stdout.read()
+                return int(result)
+
+        def assert_tempesta(self):
+                """Memory leaks detecting"""
+
+        def assert_clients(self):
+                """ Do nothing """
+
+        def create_clients(self):
+                """ Override to set desired list of benchmarks and their options. """
+                self.clients = [control.Wrk()]
+
+        def create_servers(self):
+                """ Overrirde to create needed amount of upstream servers. """
+                port = tempesta.upstream_port_start_from()
+                self.servers = [control.Nginx(listen_port=port, ka=self.backend_connections)]
+
+        def test_kmemleak(self):
+                """Detecting leaks with kmemleak"""
+                kml1 = self.read_kmemleaks()
+                self.generic_test_routine(self.config)
+                kml2 = self.read_kmemleaks()
+                if kml1 == -1 or kml2 == -1:
+                        return
```

Comment:
I think, the test must report warning if `kmemleack` is not accessible.

Reply:
fixed